project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java

```diff
@@ -27,4 +27,6 @@ Page<OntologyClass> findByLabelFulltextMatching(
 
   List<OntologyClass> findOntologyClassEntitiesByCurie(String name);
 
+  @Query(value = "SELECT DISTINCT ontology from ontology_classes;", nativeQuery = true)
+  List<String> findUniqueOntologies();
```

Comment:
A JPA interface should always return the same type. In your case I would expect a list of OntologyClasses returned.

Reply:
Unfortunately that won't be possible using the DISTINCT keyword. Alternatives would be not to use the feature or create another interface?

Reply:
you are right.